coresdk/src/backend/web_driver.cpp

```diff
@@ -85,155 +86,158 @@ namespace splashkit_lib
         curl_global_cleanup();
     }
 
-    //TODO: fix code duplication here...
-
-    sk_http_response sk_http_post(const char *host, unsigned short port, const char *body)
+    http_status_code code_to_status(long code)
     {
-        sk_http_response result = { 500, 0, NULL };
-        CURL *curl_handle;
-        CURLcode res;
-
-        // init the curl session
-        curl_handle = curl_easy_init();
+        switch (code)
+        {
+            case 200: return HTTP_STATUS_OK;
+            case 201: return HTTP_STATUS_CREATED;
+            case 204: return HTTP_STATUS_NO_CONTENT;
+            case 400: return HTTP_STATUS_BAD_REQUEST;
+            case 401: return HTTP_STATUS_UNAUTHORIZED;
+            case 403: return HTTP_STATUS_FORBIDDEN;
+            case 404: return HTTP_STATUS_NOT_FOUND;
+            case 405: return HTTP_STATUS_METHOD_NOT_ALLOWED;
+            case 408: return HTTP_STATUS_REQUEST_TIMEOUT;
+            case 500: return HTTP_STATUS_INTERNAL_SERVER_ERROR;
+            case 501: return HTTP_STATUS_NOT_IMPLEMENTED;
+            case 503: return HTTP_STATUS_SERVICE_UNAVAILABLE;
+                
+            default:
+                return static_cast<http_status_code>(code);
+        }
+    }
 
+    void _init_curl(CURL *curl_handle, const string &host, unsigned short port)
+    {
         // specify URL to get
-        curl_easy_setopt(curl_handle, CURLOPT_URL, host);
+        curl_easy_setopt(curl_handle, CURLOPT_URL, host.c_str());
 
         // set port
         curl_easy_setopt(curl_handle, CURLOPT_PORT, port);
 
         curl_easy_setopt(curl_handle, CURLOPT_FOLLOWLOCATION, 1L);
 
-        curl_easy_setopt(curl_handle, CURLOPT_POSTFIELDS, body);
-
         curl_easy_setopt(curl_handle, CURLOPT_SSL_VERIFYPEER, 0L);
 
+        // some servers don't like requests that are made without a user-agent field, so we provide one
+        curl_easy_setopt(curl_handle, CURLOPT_USERAGENT, "libcurl-agent/1.0");
+    }
+
+    struct curl_slist *_setup_curl_upload(CURL *curl_handle, const string &body)
```

Comment:
@jakerenzella When merging the twitter stuff this commit will be a merge conflict... will have to refactor again to allow custom headers.